### PR TITLE
[build] fix compiling with clang13

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80916

This check is incorrect; clang 13.1.0 doesn't exist.

### CODE REVIEW

**Comment by kit1980** [comment]
I'm not sure when the option was introduced, 13.0.1 maybe too low.
According to https://github.com/pytorch/pytorch/issues/74352, `13.1.6` is definitely has the option.

**Comment by suo** [comment]
> According to https://github.com/pytorch/pytorch/issues/74352, 13.1.6 is definitely has the option.

That refers to the Apple version. Apple's Clang 13.1.6 [corresponds](https://en.wikipedia.org/wiki/Xcode#Xcode_11.x_-_14.x_(since_SwiftUI_framework)_2) to upstream Clang 13.0.0. There is no upstream Clang 13.1.0, so this check will fail if you try to build PyTorch with Clang on Linux.

The current check as written (GTE 13.0.1) will work, but I will rewrite it to explicitly check for Apple

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -824,7 +824,8 @@ if(USE_FBGEMM)
     if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13.0.0)
       # See https://github.com/pytorch/pytorch/issues/74352
       target_compile_options(asmjit PRIVATE -Wno-deprecated-copy)
-      if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13.1.0)
+      if(("${CMAKE_CXX_COMPILER_ID}" MATCHES "AppleClang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.1.6)
+        OR ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0.0))
```
Comment (kit1980) [inline cmake/Dependencies.cmake]: This check in line 828 (without _EQUAL) is already in the previous level's if in line 824, so it's guaranteed to be true.

Comment (suo) [reply cmake/Dependencies.cmake]: yes, I leave it in for clarity (to make clear the conditional is branching on AppleClang vs. Clang)

Comment (kit1980) [reply cmake/Dependencies.cmake]: And "AppleClang" will also match "Clang", so `-Wno-unused-but-set-variable` will be added for "Apple clang version 13.0.0 (clang-1300.0.29.30)", resulting in error.

Comment (suo) [reply cmake/Dependencies.cmake]: lol you are right, I am bad

Comment (suo) [reply cmake/Dependencies.cmake]: okay it should be fixed now

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**